nodejs/eks/cluster.ts

```diff
@@ -147,6 +147,65 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         }, { parent: parent });
     }
 
+    // Create an AWS IAM Entity that will be the IAM entity which is automatically granted system:masters permissions in the cluster's RBAC configuration
+    let eksCreatorEntity: aws.iam.Role | undefined;
+    let eksCreatorEntityARN: pulumi.Input<aws.ARN> | undefined;
+    let eksCreatorEntityPolicy: aws.iam.Policy;
+    let eksCreatorEntityRolePolicyAttachment: aws.iam.RolePolicyAttachment;
+    let eksCreatorEntityProvider: aws.Provider | undefined;
+
+    if (args.useRoleForClusterCreation) {
+        eksCreatorEntity = new aws.iam.Role(`${name}-eksClusterCreatorRole`, {
```

Comment:
I assume we should be storing this `Role` as an output property of the component so that users can get it and make changes if needed.

Should the be an option to just provide this `Role` directly instead of asking to have it be created?

Reply:
Agreed, this feels like something that should be provided in an option. We tend to default to letting users create these themselves and give them to us in other areas of the cluster config.

Reply:
I think making it default and allowing users to pass in the provider or the provider-role would be the simplest way to go about it.

- Passing in only the role would create an aws provider
- Passing in both the provider and role would cause the class to create neither
- Passing in neither role nor provider would create both role and provider
- Passing in only a provider and no role would throw an error since we need the ARN of the Role to create the kubeconfig

Thoughts?